addons/binding/org.openhab.binding.tado/src/main/java/org/openhab/binding/tado/internal/TadoActionService.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.tado.internal;
+
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingRegistry;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.model.script.engine.action.ActionService;
+import org.openhab.binding.tado.TadoActions;
+
+/**
+ * Action service for {@link TadoActions}.
+ *
+ * @author Dennis Frommknecht - Iniital contribution
+ */
+public class TadoActionService implements ActionService {
+    private static ThingRegistry thingRegistry;
+
+    @Override
+    public String getActionClassName() {
+        return TadoActions.class.getCanonicalName();
+    }
+
+    @Override
+    public Class<?> getActionClass() {
+        return TadoActions.class;
+    }
+
+    public void activate() {
+    }
+
+    public void deactivate() {
+    }
+
+    public void setThingRegistry(ThingRegistry thingRegistry) {
+        TadoActionService.thingRegistry = thingRegistry;
+    }
+
+    public void unsetThingRegistry(ThingRegistry thingRegistry) {
+        TadoActionService.thingRegistry = null;
+    }
+
+    public static TadoHvacChange getHvacChange(Thing zone) {
+        return new TadoHvacChange(zone);
+    }
+
+    public static TadoHvacChange getHvacChange(String zoneThingName) {
+        Thing thing = thingRegistry.get(new ThingUID(zoneThingName));
```

Comment:
At a first glance, providing a static access to the `ThingRegistry` sounds dodgy. You are using the registry to resolve a thing via its UID. Inside your `ThingHandler` you know the thing you are responsible for and you should not need such a functionality.

Reply:
I know it's weird, but best I could find to get easy access to a zone by name from a rule.
Actions are currently not working as expected, so I would keep it at the moment and check what we need when actions in ESH were improved and we look again into it.